### PR TITLE
Tests for element reference as property function arguments

### PR DESCRIPTION
**Pull request by cdmihai**
Prompted by #1932 and https://github.com/Microsoft/msbuild/pull/1933#discussion_r110026336.
The asserts are taken from dev14's behaviour.

### CODE REVIEW

**Comment by cdmihai** [comment]
@radical

File: src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs [hunk 1]
```diff
@@ -332,6 +332,78 @@ public void ItemOperationsShouldExpandIndirectItemReferences(string projectConte
         }
 
         [Fact]
+        public void OnlyPropertyReferencesGetExpandedInPropertyFunctionArgumentsInsideIncludeAttributes()
+        {
+            var projectContent =
+@"<Project>
+        <ItemGroup>
+            <A Include=`1;$(P)` Exclude=`$(P)`/>
```
Comment (radical) [inline src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs]: Why have the `Exclude` here? I am thinking that whatever `$(P)` would evaluate to (like some incorrect value due to a bug) then it won't affect `@(A)`, so doesn't that make this a no-op, in a sense?

File: src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs [hunk 2]
```diff
@@ -332,6 +332,78 @@ public void ItemOperationsShouldExpandIndirectItemReferences(string projectConte
...
+
+        [Fact]
+        public void MetadataAndPropertyReferencesGetExpandedInPropertyFunctionArgumentsInsideMetadataElements()
+        {
+            var projectContent =
+@"<Project>
+        <ItemGroup>
+            <A Include=`1;$(P)` Exclude=`$(P)`/>
+            <B Include=`B`>
+               <M>$([System.String]::new(`%(Identity)`))</M>
```
Comment (radical) [inline src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs]: Should there be another `<M2>...` for non-built-in metadata?

File: src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs (hunk 1 above)
Comment (cdmihai) [inline src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs]: Correct, it's superfluous. I forgot it in from experimentation time.

File: src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs (hunk 2 above)
Comment (cdmihai) [inline src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs]: Wouldn't hurt, yeah. I'll add it.

**Comment by cdmihai** [comment]
@dotnet-bot test OSX Build for CoreCLR please

**Pull request merged**